test/adapt.py

```diff
@@ -92,7 +92,10 @@ def do_soup_test(self, soup_name):
         root = parse(HTML, treebuilder='soup')
         soup = root.parent
         if soup_name != 'BeautifulSoup':
-            self.ae(DOCTYPE, str(soup.contents[0]))
+            # In BS 4+, Doctype instances only store their contents. They get
+            # formatted as `<!DOCTYPE {}>` when the whole soup is serialized.
+            parsed_doctype = str(soup).split('\n', 1)[0]
```

Comment:
We could also call `soup.contents[0].output_ready()`, but that feels like more of an internal/private API, so I avoided it.